packages/react-app/src/components/pages/Bounties/Form/EditBountyForm.tsx

```diff
@@ -0,0 +1,121 @@
+import axios from '@app/utils/AxiosUtils';
+import { createRewardObject, newActivityHistory } from '@app/utils/formUtils';
+import { Button, useDisclosure } from '@chakra-ui/react';
+import { useRouter } from 'next/router';
+import React, { useCallback, useEffect, useState } from 'react';
+import { useForm } from 'react-hook-form';
+import { mutate } from 'swr';
+import DeleteBountyConfirm from './DeleteBountyConfirm';
+import BountyFormFields, { bountyFormFieldValues } from './FormFields';
+
+const getEditData = (
+	updatedBountyFormData: typeof bountyFormFieldValues,
+	exisitingBounty: BountyCollection,
+): Partial<BountyCollection> => {
+	/**
+   * Transform form data to match shape of bounty collection
+   */
+	return {
+		title: updatedBountyFormData.title,
+		description: updatedBountyFormData.description,
+		criteria: updatedBountyFormData.criteria,
+		dueAt: new Date(updatedBountyFormData.dueAt).toISOString(),
+		reward: createRewardObject(
+			updatedBountyFormData.reward,
+			updatedBountyFormData.currency
+		),
+		activityHistory: newActivityHistory(exisitingBounty.activityHistory as [], ACTIVITY.EDIT),
+	};
+};
+
+const editableValues = (bounty: BountyCollection): typeof bountyFormFieldValues => ({
+	title: bounty.title,
+	description: bounty.description,
+	reward: bounty.reward.amount?.toString() as string,
+	currency: bounty.reward.currency,
+	criteria: bounty.criteria,
+	dueAt: bounty.dueAt,
+});
+
+const EditBountyForm = ({ bounty }: { bounty: BountyCollection }): JSX.Element => {
+	const [error, setError] = useState('');
+	const [loading, setLoading] = useState(false);
+	const disclosure = useDisclosure();
+	const router = useRouter();
+	const roles = useRoles();
+	const { user } = useUser();
+	const formProps = useForm({
+		defaultValues: editableValues(bounty),
+	});
+	const { handleSubmit } = formProps;
+
+	useEffect(() => {
+		const deleted = bounty.status === BOUNTY_STATUS.DELETED;
+		const ownBounty = bounty.createdBy.discordId === user?.id;
+		const elevatedPrivs = roles.includes('admin') || roles.includes('edit-bounties') || roles.includes('delete-bounties');
+		const canEdit = elevatedPrivs || ownBounty;
+		if (deleted || !canEdit) router.push(`/${bounty._id}`);
+	}, [bounty, router]);
+
+
+	const editBounty = useCallback((editedBounty: Partial<BountyCollection>): void => {
+		/**
+         * Edit bounty is not guaranteed to have all fields, so we can
+         * Instead use the passed bounty details for customerId and _id
+         */
+		setLoading(true);
+		axios.patch<{ data: BountyCollection }>(`api/bounties/${bounty._id}?customerId=${bounty.customerId}`, editedBounty, {
+			baseURL: '/',
+		})
+			.then(({ data: res }) => {
+				mutate(`/api/bounties/${bounty._id}`, res.data);
+				router.push(`/${bounty._id}`);
+				localStorage.removeItem('cachedEdit');
+			})
+			.catch(err => {
+				console.error(err);
+				setError('There was a problem editing the bounty');
```

Comment:
Does `localStorage` persist across sessions? 

If so, may want to clear out here as well.